aws/signer/v4/v4.go

```diff
@@ -696,6 +698,81 @@ func (ctx *signingCtx) removePresign() {
 	ctx.Query.Del("X-Amz-SignedHeaders")
 }
 
+// Set ctx.Request.Host to host:port for non-standard ports, and to host for explicit standart ports
+func (ctx *signingCtx) sanitizeHost() {
+	if ctx.Request == nil || ctx.Request.URL == nil {
```

Comment:
did you run into situations where the `ctx.Request` value was nil? I think this value is required for the signer to function.  In that case i think it would be better to fail loudly instead of silently bypass the nil value.

Reply:
@jasdel, I assume it's always set, but I don't know :) Removing the check. What do you think about `ctx.Request.URL`? Should be also always set?